### ISSUE TITLE
Missing SN30/SF30 pro controllers support

### ISSUE THREAD

**Issue by superm1** [hughsie/fwupd#350]
It's these controllers: http://www.8bitdo.com/sn30pro-sf30pro/

Recently got one of these and it needs some firmware updated for some vibration problems.  I checked and didn't see the vid/pid in fwupd quirks and didn't see firmware posted at LVFS for it.  It's still at "factory" firmware right now though if any details are needed about the device in this state.

**Comment by superm1** [comment]
Oh and additional details now with testing it in bootloader mode.  It does seem to match the top quirk (bootloader: `USB\VID_2DC8&PID_5750=bootloader`).

```
06:37:29:0161 Fu  using b02f1319-c1c4-54f4-b7f2-5bb3552f6a9e for USB\VID_2DC8&PID_5750
06:37:29:0162 Fu  using 78acf201-a624-5c0b-a521-e204be14e3d5 for USB\VID_2DC8&PID_5750&REV_0200
06:37:29:0162 Fu  using 672c087de09848d9e7ee32aa1dea2fbeb8b81e6b for usb:00:01
06:37:29:0162 Fu  looking for USB quirks for FuEbitdoDevice type
06:37:29:0162 Fu  default plugin hints set to: bootloader
06:37:29:0162 Fu  looking for USB quirks for usb:00:01 device
06:37:29:0162 As  run added{2dc8:5750}
06:37:29:0221 Fu  emit added from ebitdo: 672c087de09848d9e7ee32aa1dea2fbeb8b81e6b
```

```
8Bitdo  
  DeviceId:             672c087de09848d9e7ee32aa1dea2fbeb8b81e6b
  Guid:                 b02f1319-c1c4-54f4-b7f2-5bb3552f6a9e
  Guid:                 78acf201-a624-5c0b-a521-e204be14e3d5
  Summary:              A redesigned classic game controller
  Plugin:               ebitdo
  Flags:                updatable|registered
  Vendor:               8Bitdo
  VendorId:             USB:0x2DC8
  Version:              1.03
  Icon:                 input-gaming
  Created:              2017-12-26
```

**Comment by superm1** [comment]
And in "Windows mode" (Start + X to turn on controller).
```
06:47:36:0639 Fu  using e5f6e0cf-c156-58e6-94e5-10ef2dff807c for USB\VID_045E&PID_028E
06:47:36:0639 Fu  using 5ca9a6fd-e00e-576e-9ff9-2af2cc154605 for USB\VID_045E&PID_028E&REV_0114
06:47:36:0639 Fu  using 672c087de09848d9e7ee32aa1dea2fbeb8b81e6b for usb:00:01
06:47:36:0639 Fu  looking for USB quirks for FuEbitdoDevice type
06:47:36:0639 Fu  looking for USB quirks for usb:00:01 device
06:47:36:0639 Fu  ignoring: not supported with this device
```

**Comment by superm1** [comment]
I "manually" built a CAB file by stuffing the 1.22 release into a GUID that matched it in bootloader mode.  It appeared to flash and shows the new firmware version now when I put it back in bootloader mode.

When it came back the controller booted into Nintendo Switch emulation mode (IIRC by Start + Y on controller startup).  Given the controller intentionally pretends to be a Nintendo Switch Pro controller it might be hard to tell that it's really an 8bitdo and show the firmware flash succeded.

```
8Bitdo  
  DeviceId:             672c087de09848d9e7ee32aa1dea2fbeb8b81e6b
  Guid:                 b02f1319-c1c4-54f4-b7f2-5bb3552f6a9e
  Guid:                 78acf201-a624-5c0b-a521-e204be14e3d5
  Summary:              A redesigned classic game controller
  Plugin:               ebitdo
  Flags:                updatable|registered
  Vendor:               8Bitdo
  VendorId:             USB:0x2DC8
  Version:              1.22
  Icon:                 input-gaming
  Created:              2017-12-27
```

**Comment by superm1** [comment]
This was fixed in 1.0.3 release.
